src/bolt/buffer.c

```
@@ -76,8 +77,8 @@ uint64_t buffer_index_diff
 	ASSERT(b != NULL);
 	ASSERT(a->buf == b->buf);
 
+	ASSERT(a->chunk > b->chunk || (a->chunk == b->chunk && a->offset >= b->offset));
```

Cppcheck (2.20.0)

[error] Null pointer dereference

(nullPointer)